### ISSUE TITLE
retrieval of object out side the node

### ISSUE THREAD

**Issue by vinujan59** [timohausmann/quadtree-js#6]
I am observing objects that are selected for collision which are not in the node even not in the boundry

**Comment by robblovell** [comment]
I think this manifests in the examples given. In the simple and dynamic examples, nodes that are far away from the cursor and corresponding quadtree bucket are highlighted.  Why?  They appear to be close to an x-y axis.

**Comment by robblovell** [comment]
Okay, I see what's going on. The GetIndex checks if the object is completely in the node. If it's not, the parent node is assumed. The objects that are along boundaries of the main four nodes just below the root, will always be considered to be close to all other objects.

**Comment by kfern** [comment]
This simple screenshoot show the bug.

![image](https://user-images.githubusercontent.com/1898891/75606558-45eb8280-5aee-11ea-9c19-ec9ed08322ac.png)

Any plan to solve it?

**Comment by timohausmann** [comment]
Hey, thanks for bringing this up again. Like @robblovell wrote, this behaviour stems from the tutorials design decision to store objects that don't fully fit into one quadrant on the parent node. 

However, some years passed and I get how this just seems suboptimal. So I tried the other approach: only store objects on leaf nodes (lowest level). Now if an object overlaps multiple leaf nodes, it will simply be referenced multiple times (once in each matching leaf node). This indeed achieves smaller amounts of candidates and even faster retrievement in every browser I tested (Win10 IE11, Edge, Chrome, FF). 

I put together two demos retrieving candidates from 10.000 objects with the [old algorithm 1.1.3](https://timohausmann.de/quadtree.js/test-10000-1.1.3.html) and the [new algorithm 1.2.0](https://timohausmann.de/quadtree.js/test-10000-1.2.0.html) and I'm thrilled by the improvements. 

This repository, npm registry and the online examples are updated to 1.2.0. 

I'd be happy about some feedback if this change fixes the issue or if I'm missing something. Then I will close this issue.
